src/instruments/options/finite_difference_pricer.rs

```diff
@@ -0,0 +1,491 @@
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// RustQuant: A Rust library for quantitative finance tools.
+// Copyright (C) 2023 https://github.com/avhz
+// Dual licensed under Apache 2.0 and MIT.
+// See:
+//      - LICENSE-APACHE.md
+//      - LICENSE-MIT.md
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+use crate::instruments::options::option::{ExerciseFlag, TypeFlag};
+use std::f64::consts::E;
+
+/// Finite difference object
+pub struct FiniteDifferencePricer {
+    /// Spot Price
+    pub initial_price: f64,
+    /// Strike price
+    pub strike_price: f64, 
+    /// Risk free rate
+    pub risk_free_rate: f64, 
+    /// Volatility
+    pub volatility: f64, 
+    /// Days until maturity
+    pub time_to_maturity: u16, 
```

Comment:
To be consistent, this should take a `time::Date`, see here for example: https://github.com/avhz/RustQuant/blob/main/src/instruments/options/black_scholes_merton.rs